docs/examples.asciidoc

```diff
@@ -0,0 +1,211 @@
+* <<ex-search>>
+* <<ex-scroll>>
+* <<ex-reindex>>
+
+
+[discrete]
+[[ex-index]]
+=== Indexing a document
+
+Let's index a document with the following fields: `name`, `author`, 
+`release_date`, and `page_count`:
+
+```ruby
+body = {
+  name: "The Hitchhiker's Guide to the Galaxy",
+  author: "Douglas Adams",
+  release_date: "1979-10-12",
+  page_count: 180
+}
+
+client.index(index: 'books', body: body)
+```
+
+
+[discrete]
+[[ex-get]]
+=== Getting a document
+
+You can get a document by ID:
+
+```ruby
+id = 'l7LRjXgB2_ry9btNEv_V'
+client.get(index: 'books', id: id)
+```
+
+
+[discrete]
+[[ex-update]]
+=== Updating a document
+
+Assume you have the following document:
+
+```
+book = {"name": "Foundation", "author": "Isaac Asimov", "release_date": "1951-06-01", "page_count": 224}
+```
+
+You can update the document by using the following script:
+
+```ruby
+response = client.index(index: 'books', body: book)
+id = response[‘_id’]
+client.update(index: 'books', id: id, body: { doc: { page_count: 225 } })
+```
+
+
+[discrete]
+[[ex-delete]]
+=== Deleting a document
+
+You can delete a document by ID:
+
+```ruby
+id = 'l7LRjXgB2_ry9btNEv_V'
+client.delete(index: 'books', id: id)
+```
+
+
+[discrete]
+[[ex-bulk]]
+=== Bulk requests
+
+The `bulk` operation of the client supports various different formats of the 
+payload: array of strings, header/data pairs, or the combined format where data 
+is passed along with the header in a single item in a custom `:data` key.
+
+Index several documents in one request:
+
+```ruby
+body = [
+  { index: { _index: 'books', _id: '42' } },
+  { name: "The Hitchhiker's Guide to the Galaxy", author: 'Douglas Adams', release_date: '1979-10-12', page_count: 180},
+  { index: { _index: 'books', _id: '43' } },
+  { name: 'Snow Crash', author: 'Neal Stephenson', release_date: '1992-06-01', page_count: 470 },
+  { index: { _index: 'books', _id: '44' } },
+  { name: 'Starship Troopers', author: 'Robert A. Heinlein', release_date: '1959-12-01', page_count: 335}
+]
+client.bulk(body: body)
+```
+
+Delete several documents:
+
+```ruby
+body = [
+  { delete: { _index: 'books', _id: '42' } },
+  { delete: { _index: 'books', _id: '43' } },
+]
+client.bulk(body: body)
+```
+
+As mentioned, you can perform several operations in a single request with the 
+combined format passing data in the `:data` option:
+
+```ruby
+body = [
+  { index:  { _index: 'books', _id: 45, data: { name: '1984', author: 'George Orwell', release_date: '1985-06-01', page_count: 328 } } },
+  { update: { _index: 'books', _id: 43, data: { doc: { page_count: 471 } } } },
+  { delete: { _index: 'books', _id: 44  } }
+]
+client.bulk(body: body)
+```
+
+
+[discrete]
+[[ex-search]]
+=== Searching for a document
+
+This example uses the same data that is used in the 
+https://www.elastic.co/guide/en/elasticsearch/reference/current/find-structure.html#find-structure-example-nld-json[Find structure API documentation].
+
+First, bulk index it:
+
+[source,console]
```

Comment:
```suggestion
[source,ruby]
```

Reply:
@szabosteve I added this since we use it elsewhere and in the generated documentation it generates the "Copy as curl" and "View in Console" links which we probably don't want.

Reply:
Thank you!